modules/assignment/src/test/java/com/intuit/wasabi/assignment/impl/AssignmentsImplTest.java

```diff
@@ -1670,6 +1670,46 @@ public void testGetSingleAssignmentNullAssignmentRapidExperimentCapReached() {
         verify(metadataCache, times(1)).refresh();
     }
 
+    @Test
+    public void testExperimentFetchSameNameButDeleted() {
+
+        //Create input objects
+        Application.Name appName = Application.Name.valueOf("Test");
+        Experiment.ID deletedExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.ID runningExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.Label commonExperimentLabel = Experiment.Label.valueOf("SameNamePresetTwice");
+
+        Experiment deletedExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
+        when(deletedExperiment.getID()).thenReturn(deletedExperimentId);
+        when(deletedExperiment.getEndTime().getTime()).thenReturn(new Date().getTime() + 1000000L);
+        when(deletedExperiment.getLabel()).thenReturn(commonExperimentLabel);
+        when(deletedExperiment.getState()).thenReturn(Experiment.State.DELETED);
+        when(deletedExperiment.getIsRapidExperiment()).thenReturn(false);
+
+        Experiment runningExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
+        when(runningExperiment.getID()).thenReturn(runningExperimentId);
+        when(runningExperiment.getEndTime().getTime()).thenReturn(new Date().getTime() + 1000000L);
+        when(runningExperiment.getLabel()).thenReturn(commonExperimentLabel);
+        when(runningExperiment.getState()).thenReturn(Experiment.State.RUNNING);
+        when(runningExperiment.getIsRapidExperiment()).thenReturn(false);
+
+        PrioritizedExperimentList pExpList = new PrioritizedExperimentList();
+        pExpList.addPrioritizedExperiment(PrioritizedExperiment.from(runningExperiment, 1).build());
+        Optional<PrioritizedExperimentList> prioritizedExperimentListOptional = Optional.of(pExpList);
```

Comment:
@bargenilesh Nit: Why declare this in a separate line? It is being used only once i.e. in the following line. Can we just create the optional there itself?